File: FitBook.Model/Requests/TrainingTerms/TrainingTermInsertRequest.cs
namespace FitBook.Model.Requests.TrainingTerms;

public class TrainingTermInsertRequest
{
    public DateTime StartTimeUtc { get; set; }
    public DateTime EndTimeUtc { get; set; }
    public int MaxParticipants { get; set; }
    public bool IsActive { get; set; }
    public int TrainingId { get; set; }
    public int TrainerId { get; set; }
    public int HallId { get; set; }
}

File: UI/fitbook_desktop/lib/models/requests/training_term_insert_request.dart
import 'package:json_annotation/json_annotation.dart';

import '../common/api_request_body.dart';

part 'training_term_insert_request.g.dart';

/// Mirrors `FitBook.Model.Requests.TrainingTermInsertRequest`.
@JsonSerializable()
class TrainingTermInsertRequest implements ApiRequestBody {
  TrainingTermInsertRequest({
    required this.startTimeUtc,
    required this.endTimeUtc,
    required this.maxParticipants,
    required this.isActive,
    required this.trainingId,
    required this.trainerId,
    required this.hallId,
  });

  final DateTime startTimeUtc;
  final DateTime endTimeUtc;
  final int maxParticipants;
  final bool isActive;
  final int trainingId;
  final int trainerId;
  final int hallId;

  factory TrainingTermInsertRequest.fromJson(Map<String, dynamic> json) =>
      _$TrainingTermInsertRequestFromJson(json);

  @override
  Map<String, dynamic> toJson() => _$TrainingTermInsertRequestToJson(this);
}
